### DISCUSSION TITLE
Making "role" field in User required

### DISCUSSION THREAD

**Discussion by naskio** [directus/directus#12253] [Archive]
Hello
I want to force all users in Directus to have the field "role" set to one of the available roles
How I can achieve that ?
Thank you

**Comment by rijkvanzanten** [comment]
There's no way to configure that in the platform right now. You could potentially make the role column not null directly in the database, but your mileage may vary

**Reply by naskio** [reply]
I tried to use permissions to only enable users with a role field set to be created, as follow:

![Screen Shot 2022-03-21 at 17 46 44](https://user-images.githubusercontent.com/18899702/159312508-18158a89-9886-41b0-bd8a-78a31e2dc2eb.png)

But I am not getting the expected behavior. If I try to create a new user without a role I get an error (I guess it is due to the fact that the filter is trying to access "id" and "name" of the role but it is NULL) but the user is being created (even if it shows an error message).

**Reply by rijkvanzanten** [reply]
Not sure it's the cause of that, but the rule configuration should be:

```
and
  - role->isn't null
  - role->role_name doesn't contain Administrator
```

**Reply by naskio** [reply]
Yes I tried this but you can't specify the "role" in the UI, you can only choose from its sub-fields
I may end up coding an extension (hook) to achieve that and bypass this limitation.
